### PR TITLE
Check that the RRs passed to Sign and Verify form a valid RFC2181 RRSet

### PR DESCRIPTION
**Pull request by aaronlehmann**
Add a sanity check used by RRSig's Sign and Verify functions making sure
that the records they operate on form a valid RRSet (same name, type,
and class).

Add a unit test TestInvalidRRSet that calls RRSig's Sign and Verify
methods with invalid RRSets, and makes sure the correct error is
returned.

### CODE REVIEW

File: dnssec.go [hunk 1]
```diff
@@ -221,6 +249,10 @@ func (rr *RRSIG) Sign(k PrivateKey, rrset []RR) error {
 		return ErrKey
 	}
 
+	if !isValidRRSet(rrset) {
```
Comment (miekg) [inline dnssec.go]: If you know for sure when you call Sign that you have a proper RRset this test does not add anything. I rather still leave this out of Sign itself.

File: dnssec.go [hunk 2]
```diff
@@ -205,11 +205,39 @@ func (d *DS) ToCDS() *CDS {
 	return c
 }
 
+// isValidRRSet checks if a set of RRs is a valid RRset as defined by RFC 2181.
+// This means the RRs need to have the same type, name, and class. Returns true
+// if the RR set is valid, otherwise false.
+func isValidRRSet(rrset []RR) bool {
```
Comment (miekg) [inline dnssec.go]: I think we should move this to `default.go` and name it `IsRRSet`

**Comment by miekg** [comment]
.. and sorry for the late reply, this PR slipped by me.

**Comment by aaronlehmann** [comment]
Thanks for the review. I added some commits to make these changes.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [dns] Check that the RRs passed..." ]
>Thanks for the review. I added some commits to make these changes.

cool!

**Comment by FiloSottile** [comment]
Bump

**Comment by miekg** [comment]
sorry, busy busy busy :-)

**Pull request merged**